packages/okta-auth-js/lib/token.js

```diff
@@ -752,10 +739,16 @@ function parseFromUrl(sdk, options) {
   }
 
   var url = options.url;
-  var responseMode = options.responseMode || sdk.options.responseMode || 'fragment';
+  var responseMode = options.responseMode || sdk.options.responseMode;
   var nativeLoc = sdk.token.parseFromUrl._getLocation();
   var paramStr;
 
+  if (!responseMode) {
+    // By default Authentication Code flow uses "query", Implicit flow uses "fragment"
+    // https://openid.net/specs/openid-connect-core-1_0.html#Authentication
+    responseMode = sdk.options.pkce ? 'query' : 'fragment';
+  }
```

Comment:
nit: Before defining `responseMode`, you could define `defaultResponseMode`:
```
// By default Authentication Code flow uses "query", Implicit flow uses "fragment"
// https://openid.net/specs/openid-connect-core-1_0.html#Authentication
var defaultResponseMode = sdk.options.pkce ? 'query' : 'fragment';
var responseMode = options.responseMode || sdk.options.responseMode || defaultResponseMode;
```
Arguably this manages the focus a bit better.